deepcell/applications/multiplex_segmentation.py

```diff
@@ -40,6 +40,75 @@
 from deepcell.model_zoo import PanopticNet
 
 
+import numpy as np
+
+
+def deep_watershed_subcellular(outputs, compartment='cell', min_distance=10, maxima_threshold=0.1,
+                               cell_threshold=0.1, exclude_border=False,
+                               small_objects_threshold=0):
+    """Postprocess model output to generate predictions for distinct cellular compartments
+
+    Args:
+        outputs (list): DeepWatershed model output. A list of
+            [inner_distance, outer_distance, fgbg, pixelwise], repeated for each compartment.
+            - inner_distance: Prediction for the inner distance transform.
+            - outer_distance: Prediction for the outer distance transform.
+            - fgbg: Prediction for the foregound/background transform.
+            - pixelwise: Prediction for the interior/border/background transform.
+        compartment: which cellular compartments to generate predictions for.
+            must be one of 'cell', 'nucleus', 'both'
+        min_distance (int): Minimum allowable distance between two maxima to seed cells.
+        maxima_threshold (float): Threshold for maxima used to define watershed seeds.
+        cell_threshold (float): Threshold for mask used to define allowable area for watershed.
+        exclude_border (bool): Whether to include centroid detections at the border.
+        small_objects_threshold (int): Removes objects smaller than this size.
+
+    Returns:
+        numpy.array: Uniquely labeled mask for each compartment
+
+    Raises:
+        ValueError: for invalid compartment flag
+    """
+
+    valid_compartments = ['cell', 'nucleus', 'both']
+
+    if compartment not in valid_compartments:
+        raise ValueError('Invalid compartment supplied: {}. '
+                         'Must be one of {}'.format(compartment, valid_compartments))
+
+    if compartment == 'cell':
+        label_images = deep_watershed_mibi(outputs=outputs[:4],
```

Comment:
It seems like you are assuming that outputs will always be a list of length 8 first listing cell data and then nuclear data. Is that a correct/safe assumption?

Reply:
Yeah maybe a more flexible solution is to assume that there are 2 sets of results, and instead of `outputs[:4]` we can use something like `outputs[:len(outputs)//2]`.